plugins/mac-videotoolbox/encoder.c

```diff
@@ -1005,6 +1233,107 @@ static obs_properties_t *vt_properties(void *unused, void *data)
 	return props;
 }
 
+#if ENABLE_HEVC
+static obs_properties_t *vt_properties_hevc(void *unused, void *data)
```

Comment:
As H264 and HEVC properties are so similar I think they can just be the same function, with the profile distinction being something like
```
	if (strcmp(type_data->codec, "h264") == 0) {
		obs_property_list_add_string(p, TEXT_NONE, "");
		obs_property_list_add_string(p, "baseline", "baseline");
		obs_property_list_add_string(p, "main", "main");
		obs_property_list_add_string(p, "high", "high");
#ifdef ENABLE_HEVC
	} else if (strcmp(type_data->codec, "hevc") == 0) {
		obs_property_list_add_string(p, "main", "main");
		obs_property_list_add_string(p, "main10", "main10");
		if (__builtin_available(macOS 12.3, *))
			obs_property_list_add_string(p, "main 4:2:2 10", "main42210");
#endif
	}
```